src/test/java/com/exasol/adapter/document/files/IntegrationTestSetup.java

```diff
@@ -0,0 +1,206 @@
+        final ExasolSchema adapterSchema = this.exasolObjectFactory.createSchema("ADAPTER");
+        //create a connection to Azure Blob Storage
+        this.connectionDefinition = createConnectionDefinition();
+
+        this.adapterScript = createAdapterScript(adapterSchema);
+
+        createUdf(adapterSchema);
+    }
+
+    private static void createUdf(final ExasolSchema adapterSchema) {
+        adapterSchema.createUdfBuilder("IMPORT_FROM_AZURE_BLOB_STORAGE_DOCUMENT_FILES")
+                .language(UdfScript.Language.JAVA).inputType(UdfScript.InputType.SET)
+                .parameter(PARAMETER_DOCUMENT_FETCHER, "VARCHAR(2000000)")
+                .parameter(PARAMETER_SCHEMA_MAPPING_REQUEST, "VARCHAR(2000000)")
+                .parameter(PARAMETER_CONNECTION_NAME, "VARCHAR(500)").emits()
+                .bucketFsContent(UdfEntryPoint.class.getName(), "/buckets/bfsdefault/default/" + ADAPTER_JAR).build();
+    }
+
+    private ConnectionDefinition createConnectionDefinition() {
+        final JsonObjectBuilder configJson = getConnectionConfig();
+        return createConnectionDefinition(configJson);
+    }
+    private Optional<String> getHostOverride() {
+        return this.absTestSetup.getHostOverride().map(address -> this.exasolTestSetup
+                .makeTcpServiceAccessibleFromDatabase(ServiceAddress.parse(address)).toString());
+    }
+    //creates the json connection config object
+    public JsonObjectBuilder getConnectionConfig() {
+        final JsonObjectBuilder objectBuilder = Json.createObjectBuilder();
+        if ( absTestSetup instanceof LocalAbsTestSetup){
+            //we're calling/working in the exasol database here so the connectionstring should be different
+            var defaultEndpointsProtocol = "http";
+            var accountName = "devstoreaccount1";
+            var accountKey = "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==";
+            var gho = getHostOverride();
+            //String hostMachineIp = gho.get();
+            var blobEndpoint =  "http://"+ gho.get()+ "/devstoreaccount1";
+            //var blobEndpoint =  "http://"+ hostMachineIp + ":" + ((LocalAbsTestSetup) absTestSetup).getMappedPort()+ "/devstoreaccount1";
+            var connectionString = "DefaultEndpointsProtocol="+defaultEndpointsProtocol+
+                    ";AccountName="+accountName+
+                    ";AccountKey="+ accountKey+
+                    ";BlobEndpoint="+ blobEndpoint+";";
+            return objectBuilder
+                    .add("containerName", this.absContainer.getBlobContainerName())//
+                    .add("storageAccountConnectionString", connectionString);
+        } else {
+            return objectBuilder//
+                    .add("containerName", this.absContainer.getBlobContainerName())//
+                    .add("storageAccountConnectionString", this.absTestSetup.getStorageAccountConnectionString());
```

Comment:
the two alternatives differ only in storageAccountConnectionString. If you extract a method fort that you can remove duplication of the object builder call.

Reply:
Yes I could. But this is fine imho.